tasks/synopsys-detect-task/ts/BashDetectScript.ts

```diff
@@ -0,0 +1,19 @@
+import {DetectScript} from "./DetectScript";
+import * as task from 'azure-pipelines-task-lib'
+import {ToolRunner} from "azure-pipelines-task-lib/toolrunner";
+
+export class BashDetectScript extends DetectScript {
```

Comment:
If you only care about overriding getCommands and getTool in both Bash and Powershell subclasses, I would consider making this a different heirarchy. Leave DetectScript in it's own non-subclassed heirarchy and have BashDetectScript and PowershellDetectScript implement a much simpler interface. Maybe BashTool and PowershellTool from DetectTool.